open-api/rest-catalog-open-api.yaml

```
@@ -1903,6 +1926,35 @@ components:
       schema:
         type: string
 
+    idempotency-key:
+      name: Idempotency-Key
+      in: header
+      required: false
+      schema:
+        type: string
+        format: uuid
+        minLength: 36
+        maxLength: 36
+        example: "550e8400-e29b-41d4-a716-446655440000"
+      description: |
+        Optional client-provided idempotency key for safe request retries.
+
+        When present, the server ensures no additional effects for requests that carry the same
+        Idempotency-Key within the same operation/resource scope. If a prior request with this key
+        has been finalized, the server returns an equivalent final response without re-running the
+        operation (the response body may be rebuilt from current authoritative state and need not
+        be byte-identical).
+
+        Finalization rules:
+        - Finalize & replay: 200, 201, 204, and deterministic terminal 4xx
+        - Do not finalize (not stored/replayed): 5xx, 409 (RequestInProgressException)
+
+        Key Requirements:
+        - Key format: UUID (V7 preferred)
+        - Scoped to operation type and resource path; the idempotency key must be unique per (operation type, resource path) pair (e.g., updateTable on db.t1)
+        - Catalogs may expire keys according to the advertised token lifetime.
```

Can we say this :
```suggestion
        - Catalogs MUST NOT expire keys before the end of the advertised token lifetime.
```
Mostly to avoid confusion that a catalog can expire keys early, my understanding here is, we use MAY here because catalog can give grace time as mentioned below so >= advertised token lifetime hence to explicit suggesting negation < advertised token lifetime